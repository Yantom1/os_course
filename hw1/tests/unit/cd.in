# This tests using also external command for ls, so if you dont see ls output, it might be external command bug, and not cd (but it can be cd)
# Base dir is /tmp/smash_test
pwd
cd -
cd -
cd dir1 dir2
pwd
mkdir -p dir1/dir2/dir3
pwd
ls
# /tmp/smash_test/dir1/dir2
cd dir1/dir2
pwd
ls
# /tmp/smash_test/
cd -
pwd
ls
# /tmp/smash_test/dir1/dir2
cd -
# /tmp/smash_test/dir1/dir2/dir3
cd dir3
pwd
# /tmp/smash_test/dir1/dir2
cd -
pwd
# /tmp/smash_test/dir1
cd ..
ls
# Check that failing cd wont chage last dir
pwd
cd -
pwd
cd -
pwd
cd asdasdasdasd
pwd
cd -
pwd
quit
