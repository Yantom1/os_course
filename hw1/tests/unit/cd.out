smash> /tmp/smash_test
smash> smash error: cd: OLDPWD not set
smash> smash error: cd: OLDPWD not set
smash> smash error: cd: too many arguments
smash> /tmp/smash_test
smash> smash> /tmp/smash_test
smash> dir1
my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> smash> /tmp/smash_test/dir1/dir2
smash> dir3
smash> smash> /tmp/smash_test
smash> dir1
my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> smash> smash> /tmp/smash_test/dir1/dir2/dir3
smash> smash> /tmp/smash_test/dir1/dir2
smash> smash> dir2
smash> /tmp/smash_test/dir1
smash> smash> /tmp/smash_test/dir1/dir2
smash> smash> /tmp/smash_test/dir1
smash> smash error: chdir failed: No such file or directory
smash> /tmp/smash_test/dir1
smash> smash> /tmp/smash_test/dir1/dir2
smash> 